Runtimes/Core/runtime/CMakeLists.txt

```diff
@@ -83,10 +83,10 @@ endif()
 
 # TODO: Probably worth considering putting half of these in a RuntimeConfig.h.in
 #       file rather than pushing them through macro flags.
-target_compile_definitions(swiftRuntime
+target_compile_definitions(swiftRuntimeCore
   PRIVATE
     -DSWIFT_RUNTIME
-    -DSWIFT_TARGET_LIBRARY_NAME=swiftRuntime
+    -DSWIFT_TARGET_LIBRARY_NAME=swiftRuntimeCore
```

Comment:
I believe this is the only part that actually needs changing here so that the compatibility header picks things up correctly. This is an object library that only contains C/C++, so there shouldn't be any conflict with parts from the old build. Given that the `Runtime` module depends on the overlay, it will have to be a separate "supplemental" library anyway.